lib/elixir/lib/calendar/iso.ex

```diff
@@ -102,6 +102,19 @@ defmodule Calendar.ISO do
       ]
     end
 
+  [match_ordinal_date, guard_ordinal_date, read_ordinal_date] =
+    quote do
+      [
+        <<y1, y2, y3, y4, ?-, d1, d2, d3>>,
+        y1 >= ?0 and y1 <= ?9 and y2 >= ?0 and y2 <= ?9 and y3 >= ?0 and y3 <= ?9 and y4 >= ?0 and
+          y4 <= ?9 and d1 >= ?0 and d1 <= ?9 and d2 >= ?0 and d2 <= ?9 and d3 >= ?0 and d3 <= ?9,
+        {
+          (y1 - ?0) * 1000 + (y2 - ?0) * 100 + (y3 - ?0) * 10 + (y4 - ?0),
+          (d1 - ?0) * 100 + (d2 - ?0) * 10 + (d3 - ?0)
+        }
```

Comment:
We have a function for that:
```suggestion
        {List.to_integer([y1, y2, y3, y4]), List.to_integer([d1, d2, d3])}
```